handlers.go

```diff
@@ -214,10 +209,11 @@ func (s *Server) ServeAPI() error {
 	pinning.Use(openApiMiddleware)
 	pinning.Use(s.AuthRequired(util.PermLevelUser))
 	pinning.GET("/pins", withUser(s.handleListPins))
-	pinning.POST("/pins", withUser(s.handleAddPin))
 	pinning.GET("/pins/:pinid", withUser(s.handleGetPin))
-	pinning.POST("/pins/:pinid", withUser(s.handleReplacePin))
 	pinning.DELETE("/pins/:pinid", withUser(s.handleDeletePin))
+	pinning.Use(util.JSONPayloadMiddleware)
```

Comment:
What's the difference between the previous way we handle json payload? 

If possible, please run and document tests to ensure this is backward compatible.

Reply:
I noticed many users do not read the IPFS pinning spec to know those endpoints only expect a JSON payload (content type). Ideally, we should verify our APIs only allows content types we support. This change will help communicate the error so users can make the change instead of coming to slack to figure out why their request failed.

I have tested this with the IPFS pinning compliance tool.

When standardizing our API to be more restful, both content-type and accept types should be verified if there are supported.